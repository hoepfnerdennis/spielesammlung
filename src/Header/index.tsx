import React, { memo } from 'react';
import styles from './styles.module.css';
import Search from '../Search';

const Header: React.SFC = (): JSX.Element => {
  return (
    <header className={styles.header}>
      <div className={styles.container}>
        <h1 className={styles.caption}>Spielesammlung</h1>
        <p className={styles.description}>
          Wir listen hier all unsere Spiele auf - zum Teilen, Ausleihen oder gemeinsam Spielen
        </p>
        <Search />
      </div>
    </header>
  );
};

export default memo(Header);
