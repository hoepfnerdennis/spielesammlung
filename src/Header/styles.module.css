.header {
    background-color: darkcyan;
    color: white;
}

.container {
    display: flex;
    justify-content: center;
    align-items: center;
    flex-direction: column;
    
    text-align: center;
    
    max-width: 800px;
    margin: 0 auto;
    padding: .6rem;
}

.caption {
    font-weight: 700;
    margin: 0 0 1rem 0;
    
}

.description {
    margin: 0 0 1rem 0;
}

.button {
    color: inherit;
    appearance: none;
    font: inherit;
    padding: 0;
    margin: 0;
    border: none;
    background-color: transparent;
    cursor: pointer;
}