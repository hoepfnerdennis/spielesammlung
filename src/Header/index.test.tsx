import React from 'react';
import { render } from '@testing-library/react';
import Header from './index';

// eslint-disable-next-line @typescript-eslint/explicit-function-return-type
jest.mock('../Search', () => () => 'Search');

describe('Header', () => {
  it('should render and match snapshot', () => {
    const { container } = render(<Header setFilter={jest.fn()} />);
    expect(container).toMatchSnapshot();
  });
});
