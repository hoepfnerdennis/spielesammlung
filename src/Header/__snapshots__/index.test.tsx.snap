// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header should render and match snapshot 1`] = `
<div>
  <header
    class="header"
  >
    <div
      class="container"
    >
      <h1
        class="caption"
      >
        Spielesammlung
      </h1>
      <p
        class="description"
      >
        Wir listen hier all unsere Spiele auf - zum Teilen, Ausleihen oder gemeinsam Spielen
      </p>
      Search
    </div>
  </header>
</div>
`;
